src/Http/Http.Abstractions/src/HttpContext.cs

```diff
@@ -71,4 +74,26 @@ public abstract class HttpContext
     /// Aborts the connection underlying this request.
     /// </summary>
     public abstract void Abort();
+
+    /// <inheritdoc />
+    public override string ToString()
+    {
+        return $"{Request.Method} {Request.Path.Value} {Request.ContentType}"
+            + $" Status: {Response.StatusCode} {Response.ContentType}";
+    }
+
+    private class HttpContextDebugView(HttpContext context)
+    {
+        private readonly HttpContext _context = context;
+
+        public HttpRequest Request => _context.Request;
+        public HttpResponse Response => _context.Response;
+        public ConnectionInfo Connection => _context.Connection;
+        public WebSocketManager WebSockets => _context.WebSockets;
+        public ClaimsPrincipal User => _context.User;
+        public IDictionary<object, object?> Items => _context.Items;
+        public CancellationToken RequestAborted => _context.RequestAborted;
+        public string TraceIdentifier => _context.TraceIdentifier;
+        public ISession? Session => _context.Features.Get<ISessionFeature>()?.Session;
```

Comment:
Is there a reason you can't use the property here? https://github.com/dotnet/aspnetcore/blob/c5a9c5973bf735f1bb9d611cd50c796f6c32557e/src/Http/Http.Abstractions/src/HttpContext.cs#L68

Is it because the property could throw an exception? Add a comment.

Reply:
Right, the property throws if invoked before the session middleware. I'll comment.